File: BB.Core/Model/ProductCategory.cs
using System.Collections.Generic;

namespace BB.Core.Model
{
    public class ProductCategory
    {
        public ProductCategory()
        {
            ChildrenCategories = new List<ProductCategory>();
            Products = new List<Product>();
        }

        public int ProductCategoryId { get; set; }

        public string Name { get; set; }

        public int? ParentCategoryId { get; set; }


        public virtual ProductCategory ParentCategory { get; set; }

        public virtual ICollection<ProductCategory> ChildrenCategories { get; set; }

        public virtual ICollection<Product> Products { get; set; }
    }
}

File: BB.Web/app/Model/Products/ProductCategory.ts
import p = require("../Products/Product");
import Product = p.Product;
import pp = require("../Products/ProductPicture");
import ProductPicture = pp.ProductPicture;
import c = require("../Characteristic/Characteristic");
import Characteristics = c.Characteristic;

export class ProductCategory {
    public ProductCategoryId: number;
    public Name: string;
    public ParentCategoryId?: number;
    public FacingImageId?: string;
    public FacingImage: ProductPicture;
    public ParentCategory: ProductCategory;
    public ChildrenCategories: ProductCategory[];
    public Characteristics: Characteristics[];
    public Products: Product[];

    constructor(childrenCategories: ProductCategory[], products: Product[], characteristics: Characteristics[])
    {
        this.ChildrenCategories = childrenCategories;
        this.Products = products;
        this.Characteristics = characteristics;
    }
}